src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int INCREASE_COEFFICIENT = 2;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        threshold = (int) (INCREASE_COEFFICIENT * LOAD_FACTOR);
+        nodes = new Node[INITIAL_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int hashPosition = getNodeIndex(key);
+        Node<K, V> positionNode = nodes[hashPosition];
+        if (positionNode == null) {
+            nodes[hashPosition] = newNode;
+            size++;
+            return;
+        }
+        while (positionNode != null) {
+            if (Objects.equals(key, positionNode.key)) {
+                positionNode.value = value;
+                return;
+            }
+            if (positionNode.next == null) {
+                positionNode.next = newNode;
+                size++;
+                return;
+            }
+            positionNode = positionNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> getNode = nodes[getNodeIndex(key)];
```

Comment:
Let's rename **getNode** to **node**, because **get** is a verb and it doesn't use as the first word in the names of variables